client/src/components/RelatedProducts/Cards/ProdContext.jsx

```diff
@@ -28,8 +30,47 @@ export function ProdProvider({ children }) {
     }
   }, [related]);
 
+  function convertRating(data) {
+    let totalRating = 0;
+    let avgRating = 0;
+    let totalCT = null;
+    const { ratings } = data;
+    Object.keys(ratings).forEach((key) => {
+      const currentCT = Number(ratings[key]);
+      totalRating += key * currentCT;
+      totalCT += currentCT;
+    });
+    avgRating = Math.round((totalRating / totalCT) * 10) / 10;
+    return { avgRating, totalCT };
```

Comment:
I actually calculated avg-rating in ReviewMeta and pass it in as a property.  All components have access to it with RatingProvider. See use case in ratingList line 11-13